core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -173,7 +174,7 @@ private boolean shouldRetry(SdkHttpFullResponse httpResponse, SdkException excep
         }
 
         private void executeRetry(CompletableFuture<Response<OutputT>> future) {
-            int retriesAttempted = requestCount - 2;
```

Comment:
Oh was this a bug? Are we not retrying enough times?

Reply:
No, this variable is just used for logging. It should actually be equal to `requestCount` since we only increment `requestCount` after the logging. Will update.

```
 Retryable error detected, will retry in 69ms, attempt number 1
```

Currently, it's showing `attempt number -1`